### DISCUSSION TITLE
how to add gstreamer1.0-plugins-nvv4l2camerasrc

### DISCUSSION THREAD

**Discussion by danny1209** [OE4T/meta-tegra#1093] [Q&A]
Hi all,

I put **IMAGE_INSTALL_append += " gstreamer1.0-plugins-nvv4l2camerasrc "** in local.conf and compiled.

want to use this command to stream to HDMI Display:
**gst-launch-1.0 nvv4l2camerasrc sensor-mode=0 ! "video/x-raw(memory:NVMM), format=(string)GRAY8, width=(int)1792, height=(int)1204" ! fpsdisplaysink video-sink=fakesink text-overlay=false -e -v**
but failed with this warning:
**WARNING: erroneous pipeline: no property "sensor-mode" in element "nvv4l2camerasrc0"** 

could someone tell me how to install the gstreamer plugins properly? Thanks in advance.

**Comment by kekiefer** [comment]
The `nvv4l2camerasrc` doesn't have a `sensor-mode` property, nor does it support output formats other than `UYVY`. You can use the `gst-inspect-1.0` command to see what different plugins are capable of. Perhaps it is the `nvarguscamerasrc` that you're actually wanting to use?

**Reply by danny1209** [reply]
Hi @kekiefer ,

I do mean the plugin **nvv4l2camerasrc**, it´s working in Jetpack: 
![image](https://user-images.githubusercontent.com/46848895/195358521-009ceab9-b10e-4bee-b717-0f3a80bd0eaa.png)

**Reply by danny1209** [reply]
in Jetpack:
the file is under: usr/lib/aarch64-linux-gnu/gstream-1.0/
![image](https://user-images.githubusercontent.com/46848895/195378279-fff526ff-efff-4ea9-afc8-1077caeb9652.png)

it works fine. The color is green.
 

in Yocto:
the file is under: usr/lib/gstreamer-1.0
![image](https://user-images.githubusercontent.com/46848895/195378388-efbfa214-a2e1-4e97-9203-d5ca4c46383a.png)

not green 
and nor working
![image](https://user-images.githubusercontent.com/46848895/195378485-b93906b8-4b12-4564-b021-0fa6bc16a644.png)

**Reply by kekiefer** [reply]
Well, then yes, I agree then it would be helpful to understand why there is a difference in these plugins, as meta-tegra builds nvv4l2camerasrc from the publicly available sources from NVIDIA, and that version doesn't contain a `sensor-mode` property, and only supports the `UYVY` format.

I don't know what the green means -- it's coming from bash colors, and doesn't on its own affect the way a library can behave. Maybe that green indicates it's a symlink to another plugin? Maybe it's just a permissions issue and not relevant at all?

The `nvarguscamerasrc` plugin does have a `sensor-mode` property.

**Reply by dwalkes** [reply]
> I don't know what the green means 

I think it means the executable bit is set?  I don't think that would be relevant in this case.
